src/libData/AccountData/EvmClient.cpp

```diff
@@ -32,18 +32,30 @@ void EvmClient::Init() {
 }
 
 EvmClient::~EvmClient() {
-  std::string cmdStr = "pkill " + EVM_SERVER_BINARY + " >/dev/null &";
-  LOG_GENERAL(INFO, "cmdStr: " << cmdStr);
 
+
+  std::lock_guard<std::mutex> g(m_mutexMain);
+  Json::Value _json;
+  LOG_GENERAL(DEBUG, "Call evm with die request:" << _json);
+  // call evm
   try {
-    if (!SysCommand::ExecuteCmd(SysCommand::WITHOUT_OUTPUT, cmdStr)) {
-      LOG_GENERAL(WARNING, "ExecuteCmd failed: " << cmdStr);
-    }
-  } catch (const std::exception& e) {
+    const auto oldJson = m_clients.at(0)->CallMethod("die", _json);
+  } catch (std::exception& e) {
     LOG_GENERAL(WARNING,
-                "Exception caught in SysCommand::ExecuteCmd: " << e.what());
-  } catch (...) {
-    LOG_GENERAL(WARNING, "Unknown error encountered");
+                "Caught an exception calling die " << e.what());
+    std::string cmdStr = "pkill " + EVM_SERVER_BINARY + " >/dev/null &";
+    LOG_GENERAL(INFO, "cmdStr: " << cmdStr);
+
+    try {
+      if (!SysCommand::ExecuteCmd(SysCommand::WITHOUT_OUTPUT, cmdStr)) {
+        LOG_GENERAL(WARNING, "ExecuteCmd failed: " << cmdStr);
+      }
+    } catch (const std::exception& e) {
+      LOG_GENERAL(WARNING,
+                  "Exception caught in SysCommand::ExecuteCmd: " << e.what());
+    } catch (...) {
+      LOG_GENERAL(WARNING, "Unknown error encountered");
+    }
   }
```

Comment:
Should there also be a catch(...) here? if there's an uncaught exception from a d-tor the process will terminate.